pkg/sources/sources_test.go

```diff
@@ -115,6 +115,85 @@ func TestRouter_BuildTable_CreatesRoutesForBoundSources(t *testing.T) {
 	assert.Len(t, router.GetSourceRoutes(hasNoRoutes, config.ErrorEvent), 0)
 }
 
+func TestRouter_BuildTable_CreatesRoutesWithNamespacesPresetFromKubernetesSource(t *testing.T) {
+	logger, _ := logtest.NewNullLogger()
+
+	testCases := []struct {
+		Name     string
+		Input    *config.Config
+		Expected config.Namespaces
+	}{
+		{
+			Name: "Use sources Namespaces",
+			Input: &config.Config{
+				Sources: map[string]config.Sources{
+					"k8s-events": {
+						Kubernetes: config.KubernetesSource{
+							Namespaces: config.Namespaces{
+								Include: []string{"botkube"},
+								Exclude: []string{"default"},
+							},
+							Resources: []config.Resource{
+								{
+									Name: "apps/v1/deployments",
+									Events: []config.EventType{
+										config.CreateEvent,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Expected: config.Namespaces{
+				Include: []string{"botkube"},
+				Exclude: []string{"default"},
+			},
+		},
+		{
+			Name: "Override sources Namespaces",
+			Input: &config.Config{
+				Sources: map[string]config.Sources{
+					"k8s-events": {
+						Kubernetes: config.KubernetesSource{
+							Namespaces: config.Namespaces{
+								Include: []string{"botkube"},
+								Exclude: []string{"default"},
+							},
+							Resources: []config.Resource{
+								{
+									Name: "apps/v1/deployments",
+									Namespaces: config.Namespaces{
+										Include: []string{".*"},
+									},
+									Events: []config.EventType{
+										config.CreateEvent,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Expected: config.Namespaces{
+				Include: []string{".*"},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			routes := NewRouter(nil, nil, logger).
```

Comment:
maybe we can do the same in `main.go`:
```go
		router.AddAnyBindingsByName(commGroupCfg.Slack.Channels).
			AddAnyBindingsByName(commGroupCfg.Mattermost.Channels).
			AddAnyBindings(commGroupCfg.Teams.Bindings).
			AddAnyBindingsByID(commGroupCfg.Discord.Channels)
```
but we still have a loop below, so it doesn't bring any benefit, and because of that I'm not sure if changing make sense.

Reply:
Updated the PR description.